ecs/context.go

```diff
@@ -55,45 +98,115 @@ func (h contextCreateAWSHelper) createProfile(name string) error {
 	return nil
 }
 
-func (h contextCreateAWSHelper) createContext(profile, region, description string) (interface{}, string) {
-	if profile == "default" {
-		profile = ""
+func (h contextCreateAWSHelper) createContext(c *ContextParams) (interface{}, string) {
+	if c.Profile == "default" {
+		c.Profile = ""
+	}
+	var description string
+
+	if c.CredsFromEnv {
+		if c.Description == "" {
+			description = "credentials read from environment"
+		}
+		return store.EcsContext{
+			CredentialsFromEnv: c.CredsFromEnv,
+			Profile:            c.Profile,
+			Region:             c.Region,
+		}, description
+	}
+
+	if c.Region != "" {
+		description = strings.TrimSpace(
+			fmt.Sprintf("%s (%s)", c.Description, c.Region))
 	}
-	description = strings.TrimSpace(
-		fmt.Sprintf("%s (%s)", description, region))
 	return store.EcsContext{
-		Profile: profile,
-		Region:  region,
+		Profile: c.Profile,
+		Region:  c.Region,
 	}, description
 }
 
-func (h contextCreateAWSHelper) createContextData(_ context.Context, opts ContextParams) (interface{}, string, error) {
-	profile := opts.Profile
-	region := opts.Region
-
-	profilesList, err := h.getProfiles()
+func (h contextCreateAWSHelper) selectFromLocalProfile(opts *ContextParams) error {
+	profilesList, err := getProfiles()
 	if err != nil {
-		return nil, "", err
+		return err
 	}
-	if profile != "" {
-		// validate profile
-		if profile != "default" && !contains(profilesList, profile) {
-			return nil, "", errors.Wrapf(errdefs.ErrNotFound, "profile %q", profile)
-		}
-	} else {
-		// choose profile
-		profile, err = h.chooseProfile(profilesList)
+	// choose profile
+	opts.Profile, err = h.chooseProfile(profilesList)
+	if err != nil {
+		return err
+	}
+
+	if opts.Region == "" {
+		region, isDefinedInProfile, err := getRegion(opts.Profile)
 		if err != nil {
-			return nil, "", err
+			return err
+		}
+		if isDefinedInProfile {
+			opts.Region = region
+		} else {
+			fmt.Println("No region defined in the profile. Choose the region to use.")
```

Comment:
I don't think we need this. If there is no region, we fallback to AWS default.

Reply:
Agreed. If we need to specify a region somewhere in the interactive part, we should have an optiona  flag --region to let users specify a region in scripts. 
But Maybe we don't need it, even better to only rely on AWS profiles or env vars and AWS mechanisms to get the region

Reply:
Also this will allow to simplify `getRegion(opts.Profile)`, no need to have a bool to say if the region is attached to the profile directly or not.